src/language/templates/visitors/checkparent_visitor.cpp

```diff
@@ -0,0 +1,64 @@
+/*************************************************************************
+ * Copyright (C) 2018-2019 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#include <sstream>
+#include <string>
+#include <fmt/format.h>
+#include "visitors/checkparent_visitor.hpp"
+
+namespace nmodl {
+namespace visitor {
+
+using namespace fmt::literals;
+using namespace ast;
+
+int CheckParentVisitor::check_ast(Ast* node) {
+
+    // There is no clear() in std::stack...
+    parent = nullptr;
+
+    node->accept(*this);
+
+    return 0;
+}
+
+void CheckParentVisitor::check_parent(ast::Ast* node) const {
+    if (!parent) {
+        if (is_root_with_null_parent && node->get_parent()) {
+            std::string node_type = (node == nullptr) ? std::string("nullptr") : node->get_node_type_name();
```

Comment:
```suggestion
            std::string node_type = (node == nullptr) ? "nullptr" : node->get_node_type_name();
```

Reply:
I don't see above change is made before marking conversation as resolved?